domain/model/state/modelstate.go

```diff
@@ -219,6 +219,34 @@ func (s *ModelState) GetModelMetrics(ctx context.Context) (coremodel.ModelMetric
 	}, nil
 }
 
+// GetModelCloudType returns the cloud type from a read-only model that has been
+// set in the database. If no model has been set then an error satisfying
+// [modelerrors.NotFound] is returned.
+func (s *ModelState) GetModelCloudType(ctx context.Context) (string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	m := dbReadOnlyModel{}
+	stmt, err := s.Prepare(`SELECT &dbReadOnlyModel.cloud_type FROM model`, m)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt).Get(&m)
+		if errors.Is(err, sql.ErrNoRows) {
+			return errors.New("model does not exist").Add(modelerrors.NotFound)
+		}
+		return err
+	}); err != nil {
+		return "", errors.Capture(err)
```

Comment:
I believe we should be annotating this error a bit better.

Reply:
Hi tom, thanks for the feedback, can i check when do we decide to annotate vs using errors.Capture? I used errors.Capture here as the error that occurs here is a txn error